tests/transform/test_disease_correlation.py

```diff
@@ -173,16 +162,27 @@ class TestTransformDiseaseCorrelation:
                 "allele_info": pd.DataFrame(
                     [
                         {"name": "LOAD1", "gene": "APOE4"},
+                        {"name": "LOAD1", "gene": "APOE4"},
                     ]
                 ),
             },
-            lambda output: (
-                len(output) == 1
-                and len(output[0]["results"]) == 2
-                and output[0]["results"][0] == output[0]["results"][1]
-            ),
+            lambda output: output[0]["modified_genes"] == "APOE4",
```

Comment:
I think in general it would be really helpful to add comments to the test functions and to these lambdas explaining what exactly is being tested and what the expected output should be for non-routine things. For example, I see that you are adding a duplicate `allele_info` item, but how does looking at `output[0]["modified_genes"] == "APOE4"` confirm that it has handled the duplicate properly? I would expect, for example, another check to make sure there isn't a duplicated output item too, or something.